### PR TITLE
Add checks for distribution column types not supported with upgrade

### PR DESCRIPTION
**Pull request by bhuvnesh2703**
Add checks for distribution column types not supported with upgrade

The tables with the problematic distribution column types must be either distributed randomly or on some columns which is supported. For test purposes, we distribute them randomly.

### CODE REVIEW

File: contrib/pg_upgrade/test/integration/output/unsupported_dist_coltypes.source [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+ALTER TABLE ... SET DISTRIBUTED RANDOMLY to change the distribution keys,
+and restart the upgrade.  A list of the problem columns is in the file:
+    tables_using_abstime_reltime_tinterval.txt
+
+Failure, exiting
+
+
+! /bin/cat tables_using_abstime_reltime_tinterval.txt;
+Database: upgradetest
+  public.unsupported_dist_cols1.a
```
Comment (berlin-ab) [inline contrib/pg_upgrade/test/integration/output/unsupported_dist_coltypes.source]: unsupported_dist_cols[1-4] could be renamed to articulate the reason they are unsupported.

Comment (bhuvnesh2703) [reply contrib/pg_upgrade/test/integration/output/unsupported_dist_coltypes.source]: sure

File: contrib/pg_upgrade/test/integration/input/unsupported_dist_coltypes.source [hunk 2]
```diff
@@ -0,0 +1,18 @@
+CREATE TABLE unsupported_dist_cols1(a abstime) DISTRIBUTED BY(a);
```
Comment (berlin-ab) [inline contrib/pg_upgrade/test/integration/input/unsupported_dist_coltypes.source]: I've found that the `Given/When/Then` formulations in regress tests can still be helpful to show intent for a test. Regress tests that are strictly business sometimes lack the `why` needed when reading the test at a later time.

Comment (berlin-ab) [reply contrib/pg_upgrade/test/integration/input/unsupported_dist_coltypes.source]: We've left `given/when/then` statements as comments in some regress tests.

Comment (bhuvnesh2703) [reply contrib/pg_upgrade/test/integration/input/unsupported_dist_coltypes.source]: what you say, if i write a comment, will that impart the same intent?
given - tables with unsupported distributed columns
when - upgrade check is performed
then - fails with check xxxx

Comment (berlin-ab) [reply contrib/pg_upgrade/test/integration/input/unsupported_dist_coltypes.source]: I'm saying you can be specific with types:

Given I have a table with a date as a distribution key
And [other data type]
And [other data type]
When I upgrade
Then the table with a date as a distribution key is not supported for upgrade.
And [other]

Comment (berlin-ab) [reply contrib/pg_upgrade/test/integration/input/unsupported_dist_coltypes.source]: You could even be more specific about the why in the `Then` section.

File: contrib/pg_upgrade/test/integration/gpdb6_schedule [hunk 3]
```diff
@@ -3,3 +3,4 @@ test: upgraded_exchange_partition_heap_table
 test: upgraded_partitioned_heap_table_with_differently_sized_dropped_columns
 test: upgraded_partitioned_heap_table_with_differently_aligned_dropped_columns
 test: upgraded_partitioned_heap_table_with_differently_aligned_varlen_dropped_columns
+test: upgraded_unsupported_dist_coltypes
```
Comment (dgkimura) [inline contrib/pg_upgrade/test/integration/gpdb6_schedule]: Seems like this part of the test can be parallelized with others.

Comment (bhuvnesh2703) [reply contrib/pg_upgrade/test/integration/gpdb6_schedule]: yep, will do. thanks

**Pull request merged**